mito-ai/src/Extensions/AppDeploy/DeployFilesSelector.tsx

```diff
@@ -0,0 +1,163 @@
+        .then(data => {
+          const entries: { name: string; type: string }[] = data.content.map((item: any) => ({
+            name: item.name,
+            type: item.type, // "file" or "directory"
+          }));
+          setItems(entries);
+
+          // Pre-select default files
+          const defaultFiles = new Set<string>();
+          const alwaysSelected = ['requirements.txt', 'app.py'];
+          entries.forEach(entry => {
+            if (alwaysSelected.includes(entry.name)) {
+              defaultFiles.add(entry.name);
+            }
+          });
+          setSelectedItems(defaultFiles);
+
+        })
+        .catch(err => console.error('Error fetching files/dirs:', err));
+    }
+  }, [isOpen]);
+
+  const handleCheckboxChange = (name: string) => {
+    setSelectedItems(prev => {
+      const newSet = new Set(prev);
+      if (newSet.has(name)) newSet.delete(name);
+      else newSet.add(name);
+      return newSet;
+    });
+  };
+
+  const handleSelectAll = (checked: boolean) => {
+    if (checked) setSelectedItems(new Set(items.map(i => i.name)));
+    else setSelectedItems(new Set());
+  };
+
+  const handleSubmit = () => {
+    const selectedPaths = Array.from(selectedItems);
+
+    onSubmit(selectedPaths);
+    onClose();
+  };
+
+  if (!isOpen) return null;
+
+  const allSelected = items.length > 0 && selectedItems.size === items.length;
+  const partiallySelected = selectedItems.size > 0 && selectedItems.size < items.length;
+
+  return (
+    <div className="modal-overlay">
+      <div className="modal-content">
+        <div className="modal-header">
+          <h3>Select Files or Directories</h3>
+          <button onClick={onClose} className="modal-close-button" title="Close">
+            ×
+          </button>
+        </div>
+
+        <div className="modal-body">
+          {items.length === 0 ? (
+            <p>No items found.</p>
+          ) : (
+            <>
+              {/* Select All Checkbox */}
+              <div className="select-all">
+                <label style={{ display: 'flex', alignItems: 'center', gap: '6px' }}>
+                  <input
+                    type="checkbox"
+                    checked={allSelected}
+                    ref={el => {
+                      if (el) el.indeterminate = partiallySelected;
+                    }}
+                    onChange={e => handleSelectAll(e.target.checked)}
+                  />
+                  Select All
+                </label>
+              </div>
+
+              {/* Scrollable file list */}
+              <div
+                className="file-list-scrollable"
+                style={{ maxHeight: '300px', overflowY: 'auto', marginTop: '8px' }}
+              >
+                <ul className="file-list">
+                  {items.map((item, index) => (
+                    <li key={index}>
+                      <label style={{ display: 'flex', alignItems: 'center', gap: '6px' }}>
+                        <input
```

Comment:
Since we require that the app.py and the requirements.txt file is uploaded, can we disable the checkbox in those cases so the user cannot uncheck them. We can also add a tooltip using the `title` property to say something like "Required for deploying your app" in these cases by doing something like 

```title=alwaysSelected.has(item.name) ? "Required for deploying your app" : undefined```